### PR TITLE
upd7759.cpp: Add support for mode switching.

### PR DESCRIPTION
**Pull request by wilbertpol**

### CODE REVIEW

File: src/devices/sound/upd7759.cpp [hunk 1]
```diff
@@ -647,80 +665,127 @@ WRITE_LINE_MEMBER( upd775x_device::reset_w )
...
-		logerror("upd7759_start_w: %d->%d\n", oldstart, m_start);
+	LOG_STATE("upd7759_start_w: %d->%d\n", oldstart, m_start);
 
-	/* update the stream first */
+	// update the stream first
 	m_channel->update();
 
-	/* on the rising edge, if we're idle, start going, but not if we're held in reset */
-	if (m_state == STATE_IDLE && !oldstart && m_start && m_reset)
+	// on the faling edge, if we're idle, start going, but not if we're held in reset
```
Comment (cuavas) [inline src/devices/sound/upd7759.cpp:714]: Spelling.

**Review by cuavas — COMMENTED** [review]
Please don’t turn multi-line C-style comments into sequences of single-line comments for no good reason.  There’s nothing wrong with using C-style multi-line comments, and turning them into sequences of single-line comments increases the amount of visual noise.

**Comment by angelosa** [comment]
They are also legacy comments, which I'd leave with the old C notation or remove altogether where the code is expressive enough.

```
	/* update the sample and the state */
	m_sample += upd775x_step[m_adpcm_state][data];
	m_adpcm_state += upd775x_state_table[data];
```

**Comment by wilbertpol** [comment]
> Please don’t turn multi-line C-style comments into sequences of single-line comments for no good reason. There’s nothing wrong with using C-style multi-line comments, and turning them into sequences of single-line comments increases the amount of visual noise.

I have put back the C-style comments for these multi-line comments

**Comment by wilbertpol** [comment]
> They are also legacy comments, which I'd leave with the old C notation or remove altogether where the code is expressive enough.
> 
> ```
> 	/* update the sample and the state */
> 	m_sample += upd775x_step[m_adpcm_state][data];
> 	m_adpcm_state += upd775x_state_table[data];
> ```

I have removed the comments altogether for these cases.

**Comment by Paul-Arnold** [comment]
Unfortunately this change breaks the game “brkball” in bfcobra.cpp
It has two of these devices, one master and the other slave.
One still works, the other doesn’t. Not sure which one is broke.

**Comment by wilbertpol** [comment]
> Unfortunately this change breaks the game “brkball” in bfcobra.cpp
> It has two of these devices, one master and the other slave.
> One still works, the other doesn’t. Not sure which one is broke.

Thanks, I will check it out

**Comment by Paul-Arnold** [comment]
Thanks,
I should've said the missing sounds are when the ball hits the flippers and other objects.
Adding credit should also make a very short sound.

**Comment by Paul-Arnold** [comment]
The brkball problem is due to the bfcobra driver being wrong - the driver doesn’t match the schematic for the board.
I’ll do a PR for the fix when I’m home with a decent internet connection.

**Pull request merged**